packages/plugins/src/cli/deploy.ts

```diff
@@ -222,6 +231,14 @@ export const deploy = async (
   )
   const configArtifacts = await getConfigArtifacts(initCodeWithArgsArray)
 
+  await assertNoLinkedLibraries(
```

Comment:
This comes after `await getConfigArtifacts` so that the build info cache is up to date. (The `assertNoLinkedLibraries` function uses the build info cache; more context later)